os/fs/smartfs/smartfs_smart.c

```diff
@@ -783,7 +783,11 @@ static ssize_t smartfs_write(FAR struct file *filep, const char *buffer, size_t
 				fdbg("Error %d writing sector %d data\n", ret, sf->currsector);
 				goto errout_with_semaphore;
 			}
-
```

Comment:
Line 772 `if (readwrite.count > sf->entry.datlen - sf->filepos) {`

If Line 763 sets readwrite.offset as sf->curroffset, why does 772 check for `readwrite.count > sf->entry.datlen - sf->filepos` ?

If offset to start writing is `sf->curroffset`,
then `readwrite.count = sf->entry.datlen - sf->filepos might` not match the space remaining in the sector for writing.

I think code needs to be aligned in these two places.

Reply:
in, line 763 we set write count as max size refer to fs->fs_llformat.availbytes.
But st->entry.datlen represents current file's size, so if readwrite.count + sf->filepos is greater than sf->entry.datlen, then it will overflow end of file.

Reply:
Yes exactly,
datlen-filepos can be much greater than availbytes-curroffset, so mismatch

Reply:
yes so it goes to next loop for appending, and count of readwrite become to 0.
but even if we change logic in next loop, I think performance is not changed significantly, so I will keep it.

Reply:
Understood, handled in next loop